File: source/downloads/code/LeetCode/Python/Pascal-Triangle2.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """

        dp = [1]*(rowIndex+1)

        for i in range(2, rowIndex+1):
            for j in range(1,i):
                dp[i-j] += dp[i-j-1]
        return dp

File: source/downloads/code/LeetCode/Java/Pascal-Triangle2.java
public class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> ret = new ArrayList<Integer>();
        if(rowIndex < 0)
            return ret;

        for(int i=0;i<=rowIndex;i++){
            ret.add(0,1);
            for (int j=1;j<ret.size()-1;j++){
                ret.set(j,ret.get(j+1)+ret.get(j));
            }
        } 
        return ret;
    }
}
